README.md

```diff
@@ -930,6 +930,71 @@ Output:
 
 </details>
 
+<details>
+<summary>Reflection</summary>
+
+Blueprint classes may be reflected on to inspect their views, fields, and associations. Extensions often make use of this ability.
+
+```ruby
+# A Hash of views keyed by name
+views = WidgetBlueprint.reflections
+views.keys
+=> [:default, :extended]
+
+# Hashes of fields and associations, keyed by name
+fields = views[:default].fields
+assoc = views[:default].associations
+
+# Get info about a field
+fields[:description].name
+fields[:description].display_name
+fields[:description].options
+
+# Get info about an association
+assoc[:category].name
+assoc[:category].display_name
+assoc[:category].blueprint
+assoc[:category].view
+assoc[:category].options
+```
+</details>
+
+<details>
+<summary>Extensions</summary>
+
+Blueprinter offers an extension system to hook into and modify certain behavior.
+
+```ruby
+Blueprinter.configure do |config|
+  config.extensions << MyExtension.new
+  config.extensions << OtherExtension.new
+end
+```
+
+The following class implements all extension hooks. If you have an idea for a new one, open an issue or PR.
+
+```ruby
+class ExampleExtension < Blueprinter::Extension
```

Comment:
any real-life examples that could help sparkle thought process?

Reply:
https://github.com/procore-oss/blueprinter-activerecord

Reply:
Should we add this link to the readme as well for reference?